core/src/com/unciv/models/ruleset/unit/BaseUnit.kt

```diff
@@ -334,47 +334,51 @@ class BaseUnit : INamed, INonPerpetualConstruction, ICivilopediaText {
 
         if (this.isCivilian()) return true // tiny optimization makes save files a few bytes smaller
 
-        var XP = cityConstructions.getBuiltBuildings().sumBy { it.xpForNewUnits }
+        addConstructionBonuses(unit, cityConstructions)
 
+        return true
+    }
+
+    fun addConstructionBonuses(unit: MapUnit, cityConstructions: CityConstructions) {
+        val civInfo = cityConstructions.cityInfo.civInfo
+        var XP = cityConstructions.getBuiltBuildings().sumBy { it.xpForNewUnits }
 
         for (unique in
-            cityConstructions.cityInfo.getMatchingUniques("New [] units start with [] Experience []")
-                .filter { cityConstructions.cityInfo.matchesFilter(it.params[2]) } +
-            // Deprecated since 3.15.9
+        cityConstructions.cityInfo.getMatchingUniques("New [] units start with [] Experience []")
+            .filter { cityConstructions.cityInfo.matchesFilter(it.params[2]) } +
+                // Deprecated since 3.15.9
                 cityConstructions.cityInfo.getMatchingUniques("New [] units start with [] Experience") +
                 cityConstructions.cityInfo.getLocalMatchingUniques("New [] units start with [] Experience in this city")
-            //
+        //
         ) {
             if (unit.matchesFilter(unique.params[0]))
                 XP += unique.params[1].toInt()
         }
         unit.promotions.XP = XP
```

Comment:
Sounds like an earlier `Military Units gifted from City-States start with [] XP` gets overwritten here

Reply:
It absolutely does